### PR TITLE
get cross contacts

### PR DESCRIPTION
**Pull request by PogudoTanya**

### CODE REVIEW

**Review by liubaterehova — CHANGES_REQUESTED** [review]
add function that process phonenumbers : remove "-", "()", add "+" if necessary and remove all spaces

File: src/users/users.service.ts [hunk 1]
```diff
@@ -337,4 +338,32 @@ export class UsersService extends BaseService<UserDocument> {
...
+      return `+${formattedPhoneNumber}`;
+    });
+
+    const rawListCrossContacts = await this.find(
+      { phoneNumber: { $in: rawListContactsPhone } },
+      { phoneNumber: 1, username: 1 },
+    );
+
+    return rawListCrossContacts.map((contact) => {
+      const processedListCrossContacts = rawListContacts.filter(
```
Comment (liubaterehova) [inline src/users/users.service.ts:357]: find

File: src/users/users.service.ts [hunk 2]
```diff
@@ -337,4 +338,32 @@ export class UsersService extends BaseService<UserDocument> {
     const hashNewPassword = await getHash(newPassword);
     await this.updateOne({ _id: userId }, { password: hashNewPassword });
   }
+
+  public async findCrossContacts(
+    rawListContacts: FindCrossContactsDto[],
+  ): Promise<FindCrossContactsDto[]> {
+    const rawListContactsPhone = rawListContacts.map((contact) => {
+      const formattedPhoneNumber = contact.phoneNumber.replace(/[^+\d]+/g, '');
+      if (formattedPhoneNumber.includes('+')) return formattedPhoneNumber;
```
Comment (liubaterehova) [inline src/users/users.service.ts:347]: change it

File: src/users/users.service.ts [hunk 3]
```diff
@@ -337,4 +338,32 @@ export class UsersService extends BaseService<UserDocument> {
...
+
+    const rawListCrossContacts = await this.find(
+      { phoneNumber: { $in: rawListContactsPhone } },
+      { phoneNumber: 1, username: 1 },
+    );
+
+    return rawListCrossContacts.map((contact) => {
+      const processedListCrossContacts = rawListContacts.filter(
+        (rawContact) => {
+          return contact.phoneNumber === rawContact.phoneNumber;
```
Comment (liubaterehova) [inline src/users/users.service.ts:359]: process rawContact.phoneNumber

File: src/users/users.service.ts [hunk 4]
```diff
@@ -337,4 +338,37 @@ export class UsersService extends BaseService<UserDocument> {
...
+    if (phoneNumber.startsWith('+')) return phoneNumber.replace(/[^+\d]+/g, '');
+    return `+${phoneNumber.replace(/[^+\d]+/g, '')}`;
+  }
+
+  public async findCrossContacts(
+    rawListContacts: FindCrossContactsDto[],
+  ): Promise<FindCrossContactsDto[]> {
+    const rawListContactsPhone = rawListContacts.map((contact) => {
+      const formattedPhoneNumber = this.getCorrectPhone(contact.phoneNumber);
+      return formattedPhoneNumber;
```
Comment (liubaterehova) [inline src/users/users.service.ts:352]: just `return this.getCorrectPhone(contact.phoneNumber);`

File: src/users/users.service.ts [hunk 5]
```diff
@@ -337,4 +338,37 @@ export class UsersService extends BaseService<UserDocument> {
     const hashNewPassword = await getHash(newPassword);
     await this.updateOne({ _id: userId }, { password: hashNewPassword });
   }
+
+  public getCorrectPhone(phoneNumber: string) {
+    if (phoneNumber.startsWith('+')) return phoneNumber.replace(/[^+\d]+/g, '');
+    return `+${phoneNumber.replace(/[^+\d]+/g, '')}`;
```
Comment (liubaterehova) [inline src/users/users.service.ts:344]: use ternarny operator here

File: src/users/users.service.ts [hunk 6]
```diff
@@ -337,4 +338,37 @@ export class UsersService extends BaseService<UserDocument> {
...
+      { phoneNumber: { $in: rawListContactsPhone } },
+      { phoneNumber: 1, username: 1 },
+    );
+
+    return rawListCrossContacts.map((contact) => {
+      const processedListCrossContacts = rawListContacts.find((rawContact) => {
+        return (
+          contact.phoneNumber === this.getCorrectPhone(rawContact.phoneNumber)
+        );
+      });
```
Comment (liubaterehova) [inline src/users/users.service.ts:365]: without return 
```
const processedListCrossContacts = rawListContacts.find(
        (rawContact) =>
          contact.phoneNumber === this.getCorrectPhone(rawContact.phoneNumber),
      );
```

**Review by liubaterehova — COMMENTED** [review]
Please merge your commits in one commit, fix my issues

File: src/users/users.service.ts [hunk 7]
```diff
@@ -337,4 +338,37 @@ export class UsersService extends BaseService<UserDocument> {
     const hashNewPassword = await getHash(newPassword);
     await this.updateOne({ _id: userId }, { password: hashNewPassword });
   }
+
+  public getCorrectPhone(phoneNumber: string) {
+    if (phoneNumber.startsWith('+')) return phoneNumber.replace(/[^+\d]+/g, '');
+    return `+${phoneNumber.replace(/[^+\d]+/g, '')}`;
+  }
+
+  public async findCrossContacts(
```
Comment (apsevkin) [inline src/users/users.service.ts:347]: remove async

File: src/users/users.service.ts [hunk 8]
```diff
@@ -337,4 +338,37 @@ export class UsersService extends BaseService<UserDocument> {
...
+    rawListContacts: FindCrossContactsDto[],
+  ): Promise<FindCrossContactsDto[]> {
```
Comment (apsevkin) [inline src/users/users.service.ts:349]: should return FindCrossContactsDto[] without Promise

**Review by liubaterehova — APPROVED** [review]

**Pull request merged**